src/Microsoft.AspNetCore.ResponseCompression/BodyWrapperStream.cs

```diff
@@ -214,6 +214,11 @@ private void OnWrite()
                     var compressionProvider = ResolveCompressionProvider();
                     if (compressionProvider != null)
                     {
+                        if (!HeaderUtilities.ContainsCacheDirective(HeaderNames.Vary, HeaderNames.AcceptEncoding))
```

Comment:
Actually what I said previously isn't accurate, the logic in HeaderUtilities is specific for cache directives. This is because it only handles the cases where the header values are in the format `valueName=<someValue>, ...` which doesn't generalize to other headers such as Accept where the headers can look like `valueName;q=<qValue>, ...`. For the use case here, the layout of the header value is even simpler since it is a comma separated list of tokens. I'll add a separate helper function.